data_io/data_io.py

```diff
@@ -482,46 +537,27 @@ def write(
         tempdir="s3a://tfsds-lsg-test/ingestion/redshift_temp",
         mode='default'
       )
-
-    writer = data.write
-
-    # We'll set the default read options then override with
-    # whatever the user wants
-    write_options = default_write_options
-  
-    for option, value in kwargs.items():
-
-      write_options[option] = value
-
-    # Set format and mode
-    writer = (
-      writer
-      .format(write_options['format'])
-      .mode(write_options['mode'])
+    
+    writer = super().set_writer(
```

Comment:
Why `super()` instead of `self`?

Reply:
At some point while making these updates, using `self` was throwing an error. But, I replaced all the `super().` calls with `self.` and there are no errors. Must have been related to some other unknown bug I introduced that was fixed along the way.